mscore/preferences.cpp

```diff
@@ -1424,6 +1442,13 @@ void PreferenceDialog::apply()
 
 #ifdef USE_PORTMIDI
       prefs.portMidiInput = portMidiInput->currentText();
+      prefs.portMidiOutput = portMidiOutput->currentText();
+      if (static_cast<PortMidiDriver*>(static_cast<Portaudio*>(seq->driver())->mididriver())->isSameCoreMidiIacBus(prefs.portMidiInput, prefs.portMidiOutput)) {
+            QMessageBox msgBox;
+            msgBox.setText("Warning: You used the same CoreMIDI IAC bus for input and output.  This will cause problematic loopback of MuseScore's output midi messages, since they will be sent back in to MuseScore as input, causing confusion.  To avoid this problem, use Mac OS's Audio MIDI Setup to create a dedicated virtual port for MuseScore's Midi output, restart MuseScore, and then select that port as output.  Then other programs may use that deidcated virtual port to receive MuseScore's midi input.");
```

Comment:
I can't test this cause I don't have a mac, but this is what the modal popup warning would look like:

![screenshot 215](https://cloud.githubusercontent.com/assets/6502474/26158339/8d5b90ce-3ae9-11e7-8d7b-b72ba027e1ad.png)

I admit it is a bit ugly and wordy, but I'd rather have the user see this and fix it themselves rather than posting to the forums for help.